File: src/app/models/Departement.ts
import {Pays} from "./Pays";

export class Departement {
  id: number;
  libelle:	string;
  createBy:	string;
  deleteBy:	string;
  pays: Pays;

  constructor(id: number, libelle: string, createBy: string, deleteBy: string, pays: Pays) {
    this.id = id;
    this.libelle = libelle;
    this.createBy = createBy;
    this.deleteBy = deleteBy;
    this.pays = pays;
  }
}

File: sgifd-backend/src/main/java/com/sgifdbackend/demo/parametrage/entites/Departement.java
package com.sgifdbackend.demo.parametrage.entites;

import com.sgifdbackend.demo.entitiesBase.EntityBaseBean;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "t_Departement")
public class Departement extends EntityBaseBean implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String libelle;
   

    @ManyToOne
    private Pays pays;

    public Departement() {
    }


    public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}


	public String getLibelle() {
        return libelle;
    }

    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }


	public Pays getPays() {
		return pays;
	}


	public void setPays(Pays pays) {
		this.pays = pays;
	}

}
